### PR TITLE
Overwrite marginal() method to properly handle measurement noise (e.g……., uncertainty)

### PR DESCRIPTION
**Pull request by djross22**
From my testing, the noise parameter was never getting passed to the _shaped_noise_covar() method. So, the measurement uncertainty was not taken into account in the calculations of the loss.

Looking at the code, I think the simplest change needed to handle the measurement noise is to overwrite the marginal() method in the lantern MultitaskGaussianLikelihood class, and explicitely pass the noise parameter to _shaped_noise_covar() (if it exists).

### CODE REVIEW

**Comment by djross22** [comment]
Also, to make this code work, the 'noise' parameter has to be explicitly passed to the loss function during the training loop.
So, instead of:
```
for btch in loader:
    optimizer.zero_grad()
    yhat = model(btch[0])
    lss = loss(yhat, *btch[1:])
    ...
```
you need something like:
```
for btch in loader:
    optimizer.zero_grad()
    yhat = model(btch[0])
    noise = btch[2]
    lss = loss(yhat, btch[1], noise=noise)
    ...
```

**Comment by ptonner** [comment]
I tried briefly to figure out what, if anything, changed in the upstream GP library that may have caused the issue - no dice. Regardless, you patch looks good
